ui/src/loading.tsx

```diff
@@ -1,26 +1,65 @@
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
 import CircularProgress from "@mui/material/CircularProgress";
-import Grid from "@mui/material/Grid";
-import React, { useEffect, useRef, useState } from "react";
+import Typography from "@mui/material/Typography";
+import React, { ReactNode, useEffect, useRef, useState } from "react";
 
-export default function Loading() {
+type State = {
+  error?: Error;
+  isPending: boolean;
+  reload: () => void;
+};
+
+type Props = {
+  state?: State;
```

Comment:
I would avoid having a prop called `state`. I realize you probably don't mean to imply "React state" with this naming, I would guess most readers make that assumption. In general I would not recommend using the term state in the vicinity of a React component unless it's actually referring to React state.

Something like "status" maybe.

Reply:
Done.